go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1986,35 +1987,53 @@ func commonFlows(subnets []*net.IPNet, bridge *bridgeConfiguration, isPodNetwork
 	}
 
 	if ofPortPhys != "" {
-		if config.Gateway.DisableSNATMultipleGWs || isPodNetworkAdvertised {
+		if config.Gateway.DisableSNATMultipleGWs || isPodNetworkAdvertised || isUDNNetworkAdvertised {
 			// table 1, traffic to pod subnet go directly to OVN
-			output := defaultNetConfig.ofPortPatch
-			if isPodNetworkAdvertised && config.Gateway.Mode == config.GatewayModeLocal {
-				// except if advertised through BGP, go to kernel
-				// TODO: MEG enabled pods should still go through the patch port
-				// but holding this until
-				// https://issues.redhat.com/browse/FDP-646 is fixed, for now we
-				// are assuming MEG & BGP are not used together
-				output = ovsLocalPort
-			}
-			for _, clusterEntry := range config.Default.ClusterSubnets {
-				cidr := clusterEntry.CIDR
-				ipv := getIPv(cidr)
-				dftFlows = append(dftFlows,
-					fmt.Sprintf("cookie=%s, priority=15, table=1, %s, %s_dst=%s, "+
-						"actions=output:%s",
-						defaultOpenFlowCookie, ipv, ipv, cidr, output))
-			}
-			if output == defaultNetConfig.ofPortPatch {
-				// except node management traffic
-				for _, subnet := range subnets {
-					mgmtIP := util.GetNodeManagementIfAddr(subnet)
-					ipv := getIPv(mgmtIP)
+			for netName, netConfig := range bridge.netConfig {
+				if isPodNetworkAdvertised && netName != types.DefaultNetworkName {
+					continue
+				} else if isUDNNetworkAdvertised && netName == types.DefaultNetworkName {
+					continue
+				}
+				output := netConfig.ofPortPatch
+				if (isPodNetworkAdvertised || isUDNNetworkAdvertised) && config.Gateway.Mode == config.GatewayModeLocal {
```

Comment:
How do you know to which network does the `isUDNNetworkAdvertised` apply to? Have you tested with multiple UDNs where only a subset of them is advertised? It looks like just as `bridge.netConfig` is a map with different networks, we should also have a map to store `isUDNNetworkAdvertised` per network. I guess we can use `bridge.netConfig` as well for this.